linera-chain/src/data_types.rs

```diff
@@ -614,16 +614,19 @@ pub struct ProposalContent {
 }
 
 impl BlockProposal {
-    pub fn new_initial(round: Round, block: ProposedBlock, secret: &AccountSecretKey) -> Self {
+    pub fn new_initial(round: Round, block: ProposedBlock, signer: &Box<dyn Signer>) -> Self {
```

Comment:
Clippy is right, I'd even say `&impl Signer`.

Reply:
Done, albeit force-pushed.